lib/object/has-property.js

```diff
@@ -5,22 +5,39 @@
  * @author Eric Satterwhite
  **/
 
-module.exports = function hasOwnProperty(obj, prop) {
-  return Object.prototype.hasOwnProperty.call(obj, prop)
+module.exports = function hasOwnProperty(obj, string = '', sep = '.') {
+  /* eslint-disable-next-line no-eq-null */
+  if (obj == null) return false
+
+  const parts = string.split(sep)
+  let ret = obj
+  /* eslint-disable-next-line no-unused-vars */
+  const last = parts.pop()
+  let prop
+  while ((prop = parts.shift())) {
+    ret = ret[prop]
```

Comment:
So this kind of break the hasOwnProperty rule. Property subscripting looks up the prototype chain.
If you really need to know if only the constructed object has a property, you would need to use `hasOwnProperty` here as well

Reply:
Done!